samples/AnomalyDetector/src/Generated/Models/AnomalyState.Serialization.cs

```diff
@@ -32,15 +32,22 @@ void IJsonModel<AnomalyState>.Write(Utf8JsonWriter writer, ModelReaderWriterOpti
             if (Optional.IsDefined(Value))
             {
                 writer.WritePropertyName("value"u8);
-                writer.WriteObjectValue(Value);
+                ((IJsonModel<AnomalyValue>)Value).Write(writer, options);
             }
             if (Optional.IsCollectionDefined(Errors))
             {
                 writer.WritePropertyName("errors"u8);
                 writer.WriteStartArray();
                 foreach (var item in Errors)
                 {
-                    writer.WriteObjectValue(item);
+                    if (item != null)
+                    {
+                        ((IJsonModel<ErrorResponse>)item).Write(writer, options);
+                    }
+                    else
+                    {
+                        writer.WriteNullValue();
+                    }
                 }
```

Comment:
this is the second kind of the change: for elements of collections
We need to check null here because there is a chance that the value is null (csharp language allows that)